video/out/wayland_common.c

```diff
@@ -151,12 +151,12 @@ static void pointer_handle_motion(void *data, struct wl_pointer *pointer,
                                   uint32_t time, wl_fixed_t sx, wl_fixed_t sy)
 {
     struct vo_wayland_state *wl = data;
-    if (!wl->prev_fullscreen && wl->fullscreen) {
-        wl->prev_fullscreen = wl->fullscreen;
+    if (!wl->prev_fullscreen && wl->vo_opts->fullscreen) {
+        wl->prev_fullscreen = wl->vo_opts->fullscreen;
```

Comment:
I don't think replacing `wl->fullscreen` with `wl->vo_opts->fullscreen` is correct (here or in the other places). Assuming I understand `vo_opts` correctly, it's just merely a replacement for accessing `vo->opts` directly. `wl->fullscreen` is meant to tell you what the state of the window actually is not its option.